binary-install/src/lib.rs

```diff
@@ -270,3 +260,82 @@ fn curl(url: &str) -> Result<Vec<u8>, Error> {
         )
     }
 }
+
+fn hashed_dirname(url: &str, name: &str) -> String {
+    let mut hasher = SipHasher13::new();
+    url.hash(&mut hasher);
+    let result = hasher.finish();
+    let hex = hex::encode(&[
+        (result >> 0) as u8,
+        (result >> 8) as u8,
+        (result >> 16) as u8,
+        (result >> 24) as u8,
+        (result >> 32) as u8,
+        (result >> 40) as u8,
+        (result >> 48) as u8,
+        (result >> 56) as u8,
+    ]);
+    format!("{}-{}", name, hex)
+}
+
+#[test]
+fn it_returns_same_hash_for_same_name_and_url() {
```

Comment:
Any tests that need access to private functions/types should be inside a `cfg`ed module like this:

```rust
#[cfg(test)]
mod tests {
    user super::*;

    // ...
}
```

This way they aren't included at all in non-test builds.

Reply:
Right, missed that. Thanks!